lib/origen/registers/bit_collection.rb

```diff
@@ -509,6 +509,7 @@ def overlay_str
       def write!(value = nil, options = {})
         value, options = nil, value if value.is_a?(Hash)
         write(value, options) if value
+        yield @reg if block_given?
```

Comment:
Think these should be yield self and not @reg.

They are effectively the same when operating on a register, but this update does also support the same kind of operation on a subset of bits:

~~~ruby
some_reg.some_bits.write! do |b|
  b[1].write(1)
  b[4].write(1)
end
~~~

By yielding @reg the internal object would be the register and not the subset of bits and any relative position references like above would be off.

Reply:
Hmm - that does not work...

Reply:
TypeError: no implicit conversion of Symbol into Integer
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/registers/bit_collection.rb:48:in `fetch'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/registers/bit_collection.rb:48:in `block in []'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/registers/bit_collection.rb:47:in `each'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/registers/bit_collection.rb:47:in `[]'
	from (origen):5:in `block in irb_binding'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/registers/bit_collection.rb:512:in `write!'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/registers/reg.rb:1228:in `method_missing'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/registers.rb:249:in `method_missing'
	from (origen):4
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/commands/interactive.rb:79:in `<module:Origen>'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/commands/interactive.rb:11:in `<top (required)>'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/commands.rb:233:in `require'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/lib/origen/commands.rb:233:in `<top (required)>'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/bin/origen:172:in `require'
	from /proj/mcusoc_kodiak/users/chrisn/origen/origen/bin/origen:172:in `<top (required)>'
	from ./lbin/origen:17:in `load'

Reply:
The example as I wrote it above you mean, or something else?

Reply:
Using self instead of @reg results in error above

Reply:
Just pushed an update that fixes it.
I though `<bit collection>.bits` worked, but it has to be `<reg>.bits`